protocol/x/vault/keeper/shares.go

```diff
@@ -0,0 +1,34 @@
+package keeper
+
+import (
+	"cosmossdk.io/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/vault/types"
+)
+
+// GetTotalShares gets TotalShares for a vault.
+func (k Keeper) GetTotalShares(
+	ctx sdk.Context,
+	vaultId types.VaultId,
+) (val types.NumShares, exists bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.TotalSharesKeyPrefix))
+
+	b := store.Get(vaultId.ToStateKey())
+	if b == nil {
+		return val, false
```

Comment:
how is `val` being returned here without being initialized? Is `val` being in the return of this function also initialize it? Does that also mean `exists` is initialized?

Reply:
Yes, that's correct. See [here](https://go.dev/tour/basics/7#:~:text=Go's%20return%20values%20may%20be,returns%20the%20named%20return%20values.) for more details.